easy-ui-react/src/DataGrid/useExpandedRow.ts

```diff
@@ -0,0 +1,177 @@
+import { useResizeObserver } from "@react-aria/utils";
+import {
+  CSSProperties,
+  MutableRefObject,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { Node, TableState } from "react-stately";
+import { getComponentToken } from "../utilities/css";
+import { EXPAND_COLUMN_KEY } from "./constants";
+
+/**
+ * Manages the size and positioning of the expanded row content with
+ * minimal-to-no jank. It uses an invisible placeholder box to set and get the
+ * appropriate bounding client rectangle representative of the expanded row
+ * content box, before applying it to the visible expanded row.
+ */
```

Comment:
**question**: so this computes the invisible box before actually applying it to the expanded row? this is probably a silly question and i might be missing something, but why is that necessary?

Reply:
your comment made me realize that we don't need the invisible box after all. after creating this implementation, i refactored the DataGrid to use a separate scroll container from the grid itself. now we can update the box in one render cycle, before it required two which needed a second box to manage that render cycle delay. thanks lol!